packages/grafana-data/src/transformations/matchers/nameMatcher.ts

```diff
@@ -4,32 +4,40 @@ import { FieldMatcherInfo, FrameMatcherInfo } from '../../types/transformations'
 import { stringToJsRegex } from '../../text/string';
 import { getFieldDisplayName } from '../../field/fieldState';
 
-// General Field matcher
-const fieldNameMacher: FieldMatcherInfo<string> = {
+export interface FieldNameMatcherOptions {
+  pattern: string;
+  frame?: DataFrame;
+  series?: DataFrame[];
+}
+
+const fieldNameMatcher: FieldMatcherInfo<FieldNameMatcherOptions> = {
   id: FieldMatcherID.byName,
   name: 'Field Name',
   description: 'match the field name',
-  defaultOptions: '/.*/',
+  defaultOptions: {
+    pattern: '/.*/',
+  },
 
-  get: (pattern: string) => {
+  get: (options: FieldNameMatcherOptions) => {
+    const { pattern, frame, series } = options;
     let regex = new RegExp('');
     try {
       regex = stringToJsRegex(pattern);
     } catch (e) {
       console.error(e);
     }
     return (field: Field) => {
-      return regex.test(getFieldDisplayName(field) ?? '');
+      return regex.test(getFieldDisplayName(field, frame, series) ?? '');
     };
   },
 
-  getOptionsDisplayText: (pattern: string) => {
-    return `Field name: ${pattern}`;
+  getOptionsDisplayText: (options: FieldNameMatcherOptions) => {
+    return `Field name: ${options.pattern}`;
   },
 };
 
 // General Field matcher
-const frameNameMacher: FrameMatcherInfo<string> = {
+const frameNameMatcher: FrameMatcherInfo<string> = {
```

Comment:
do we need the same options for this?  `string` will fail in the same way, no?

Reply:
hmm, not sure. Good feedback! Will have a look at that one once the field matcher is in place.